algorithms/src/sorting/impl/Kokkos_SortByKeyImpl.hpp

```diff
@@ -87,7 +87,19 @@ static_assert_is_admissible_to_kokkos_sort_by_key(const ViewType& /* view */) {
                 "LayoutRight, LayoutLeft or LayoutStride.");
 }
 
+// For the fallback sort_by_key implementation we might need to execute
+// Kokkos::sort on the host. This type trait determines at compile-time where we
```

Comment:
> we might need to execute Kokkos::sort on the host

"We need to consider if Kokkos::sort defers to the fallback implementation that copies the array to the host and uses std::sort.  The value is determined by the implementation choices in the `impl/Kokkos_SortImpl.hpp` which mean they need to be kept in sync (sigh...)"